### ISSUE TITLE
Failing Preview tests on PHP7

### ISSUE THREAD

**Issue by rullzer** [nextcloud/server#331]
The fancy new PHP7 CI runs (https://github.com/nextcloud/server/pull/330) have failing tests (preview).

See: https://drone.weasel.rocks/nextcloud/server/316

Most of them seem as an off by 1 error somewhere. So maybe php7 does rounding somewhat different for some reason.

Important part:

```

There were 38 failures:

1) Test\PreviewTest::testCreateMaxAndNormalPreviewsAtFirstRequest with data set #16 (1, -427, -507, false, false)
/test-preview-user1/thumbnails/2718/756-537-max.png 

Failed asserting that false matches expected true.

/drone/src/github.com/nextcloud/server/tests/lib/PreviewTest.php:343

2) Test\PreviewTest::testCreateMaxAndNormalPreviewsAtFirstRequest with data set #17 (1, 554, 405, false, false)
/test-preview-user1/thumbnails/2728/756-537-max.png 

Failed asserting that false matches expected true.

/drone/src/github.com/nextcloud/server/tests/lib/PreviewTest.php:343

3) Test\PreviewTest::testCreateMaxAndNormalPreviewsAtFirstRequest with data set #18 (1, -654, 714, false, false)
/test-preview-user1/thumbnails/2738/756-537-max.png 

Failed asserting that false matches expected true.

/drone/src/github.com/nextcloud/server/tests/lib/PreviewTest.php:343

4) Test\PreviewTest::testCreateMaxAndNormalPreviewsAtFirstRequest with data set #19 (1, 687, -782, false, false)
/test-preview-user1/thumbnails/2748/756-537-max.png 

Failed asserting that false matches expected true.

/drone/src/github.com/nextcloud/server/tests/lib/PreviewTest.php:343

5) Test\PreviewTest::testCreateMaxAndNormalPreviewsAtFirstRequest with data set #20 (1, -580, -228, false, true)
/test-preview-user1/thumbnails/2758/756-537-max.png 

Failed asserting that false matches expected true.

/drone/src/github.com/nextcloud/server/tests/lib/PreviewTest.php:343

6) Test\PreviewTest::testCreateMaxAndNormalPreviewsAtFirstRequest with data set #21 (1, 714, 438, false, true)
/test-preview-user1/thumbnails/2768/756-537-max.png 

Failed asserting that false matches expected true.

/drone/src/github.com/nextcloud/server/tests/lib/PreviewTest.php:343

7) Test\PreviewTest::testCreateMaxAndNormalPreviewsAtFirstRequest with data set #22 (1, -505, 590, false, true)
/test-preview-user1/thumbnails/2778/756-537-max.png 

Failed asserting that false matches expected true.

/drone/src/github.com/nextcloud/server/tests/lib/PreviewTest.php:343

8) Test\PreviewTest::testCreateMaxAndNormalPreviewsAtFirstRequest with data set #23 (1, 630, -337, false, true)
/test-preview-user1/thumbnails/2788/756-537-max.png 

Failed asserting that false matches expected true.

/drone/src/github.com/nextcloud/server/tests/lib/PreviewTest.php:343

9) Test\PreviewTest::testCreateMaxAndNormalPreviewsAtFirstRequest with data set #24 (1, -541, -799, true, false)
/test-preview-user1/thumbnails/2798/756-537-max.png 

Failed asserting that false matches expected true.

/drone/src/github.com/nextcloud/server/tests/lib/PreviewTest.php:343

10) Test\PreviewTest::testCreateMaxAndNormalPreviewsAtFirstRequest with data set #25 (1, 595, 398, true, false)
/test-preview-user1/thumbnails/2807/756-537-max.png 

Failed asserting that false matches expected true.

/drone/src/github.com/nextcloud/server/tests/lib/PreviewTest.php:343

11) Test\PreviewTest::testCreateMaxAndNormalPreviewsAtFirstRequest with data set #26 (1, -672, 279, true, false)
/test-preview-user1/thumbnails/2816/756-537-max.png 

Failed asserting that false matches expected true.

/drone/src/github.com/nextcloud/server/tests/lib/PreviewTest.php:343

12) Test\PreviewTest::testCreateMaxAndNormalPreviewsAtFirstRequest with data set #27 (1, 459, -722, true, false)
/test-preview-user1/thumbnails/2825/756-537-max.png 

Failed asserting that false matches expected true.

/drone/src/github.com/nextcloud/server/tests/lib/PreviewTest.php:343

13) Test\PreviewTest::testCreateMaxAndNormalPreviewsAtFirstRequest with data set #28 (1, -441, -734, true, true)
/test-preview-user1/thumbnails/2834/756-537-max.png 

Failed asserting that false matches expected true.

/drone/src/github.com/nextcloud/server/tests/lib/PreviewTest.php:343

14) Test\PreviewTest::testCreateMaxAndNormalPreviewsAtFirstRequest with data set #29 (1, 430, 619, true, true)
/test-preview-user1/thumbnails/2843/756-537-max.png 

Failed asserting that false matches expected true.

/drone/src/github.com/nextcloud/server/tests/lib/PreviewTest.php:343

15) Test\PreviewTest::testCreateMaxAndNormalPreviewsAtFirstRequest with data set #30 (1, -334, 397, true, true)
/test-preview-user1/thumbnails/2852/756-537-max.png 

Failed asserting that false matches expected true.

/drone/src/github.com/nextcloud/server/tests/lib/PreviewTest.php:343

16) Test\PreviewTest::testCreateMaxAndNormalPreviewsAtFirstRequest with data set #31 (1, 367, -787, true, true)
/test-preview-user1/thumbnails/2861/756-537-max.png 

Failed asserting that false matches expected true.

/drone/src/github.com/nextcloud/server/tests/lib/PreviewTest.php:343

17) Test\PreviewTest::testSecondPreviewsGetCachedMax with data set #16 (1, -443, -457, false, false)
Failed asserting that two strings are identical.
--- Expected
+++ Actual
@@ @@
-thumbnails/3327/756-537-max.png
+thumbnails/3327/756-538-max.png

/drone/src/github.com/nextcloud/server/tests/lib/PreviewTest.php:408

18) Test\PreviewTest::testSecondPreviewsGetCachedMax with data set #17 (1, 672, 582, false, false)
Failed asserting that two strings are identical.
--- Expected
+++ Actual
@@ @@
-thumbnails/3337/756-537-max.png
+thumbnails/3337/756-538-max.png

/drone/src/github.com/nextcloud/server/tests/lib/PreviewTest.php:408

19) Test\PreviewTest::testSecondPreviewsGetCachedMax with data set #18 (1, -309, 463, false, false)
Failed asserting that two strings are identical.
--- Expected
+++ Actual
@@ @@
-thumbnails/3347/756-537-max.png
+thumbnails/3347/756-538-max.png

/drone/src/github.com/nextcloud/server/tests/lib/PreviewTest.php:408

20) Test\PreviewTest::testSecondPreviewsGetCachedMax with data set #19 (1, 442, -504, false, false)
Failed asserting that two strings are identical.
--- Expected
+++ Actual
@@ @@
-thumbnails/3357/756-537-max.png
+thumbnails/3357/756-538-max.png

/drone/src/github.com/nextcloud/server/tests/lib/PreviewTest.php:408

21) Test\PreviewTest::testSecondPreviewsGetCachedMax with data set #20 (1, -419, -330, false, true)
Failed asserting that two strings are identical.
--- Expected
+++ Actual
@@ @@
-thumbnails/3367/756-537-max.png
+thumbnails/3367/756-538-max.png

/drone/src/github.com/nextcloud/server/tests/lib/PreviewTest.php:408

22) Test\PreviewTest::testSecondPreviewsGetCachedMax with data set #21 (1, 785, 391, false, true)
Failed asserting that two strings are identical.
--- Expected
+++ Actual
@@ @@
-thumbnails/3377/756-537-max.png
+thumbnails/3377/756-538-max.png

/drone/src/github.com/nextcloud/server/tests/lib/PreviewTest.php:408

23) Test\PreviewTest::testSecondPreviewsGetCachedMax with data set #22 (1, -634, 616, false, true)
Failed asserting that two strings are identical.
--- Expected
+++ Actual
@@ @@
-thumbnails/3387/756-537-max.png
+thumbnails/3387/756-538-max.png

/drone/src/github.com/nextcloud/server/tests/lib/PreviewTest.php:408

24) Test\PreviewTest::testSecondPreviewsGetCachedMax with data set #23 (1, 305, -636, false, true)
Failed asserting that two strings are identical.
--- Expected
+++ Actual
@@ @@
-thumbnails/3397/756-537-max.png
+thumbnails/3397/756-538-max.png

/drone/src/github.com/nextcloud/server/tests/lib/PreviewTest.php:408

25) Test\PreviewTest::testSecondPreviewsGetCachedMax with data set #24 (1, -424, -657, true, false)
Failed asserting that two strings are identical.
--- Expected
+++ Actual
@@ @@
-thumbnails/3407/756-537-max.png
+thumbnails/3407/756-538-max.png

/drone/src/github.com/nextcloud/server/tests/lib/PreviewTest.php:408

26) Test\PreviewTest::testSecondPreviewsGetCachedMax with data set #25 (1, 357, 250, true, false)
Failed asserting that two strings are identical.
--- Expected
+++ Actual
@@ @@
-thumbnails/3417/756-537-max.png
+thumbnails/3417/756-538-max.png

/drone/src/github.com/nextcloud/server/tests/lib/PreviewTest.php:408

27) Test\PreviewTest::testSecondPreviewsGetCachedMax with data set #26 (1, -677, 613, true, false)
Failed asserting that two strings are identical.
--- Expected
+++ Actual
@@ @@
-thumbnails/3427/756-537-max.png
+thumbnails/3427/756-538-max.png

/drone/src/github.com/nextcloud/server/tests/lib/PreviewTest.php:408

28) Test\PreviewTest::testSecondPreviewsGetCachedMax with data set #27 (1, 320, -223, true, false)
Failed asserting that two strings are identical.
--- Expected
+++ Actual
@@ @@
-thumbnails/3437/756-537-max.png
+thumbnails/3437/756-538-max.png

/drone/src/github.com/nextcloud/server/tests/lib/PreviewTest.php:408

29) Test\PreviewTest::testSecondPreviewsGetCachedMax with data set #28 (1, -780, -669, true, true)
Failed asserting that two strings are identical.
--- Expected
+++ Actual
@@ @@
-thumbnails/3447/756-537-max.png
+thumbnails/3447/756-538-max.png

/drone/src/github.com/nextcloud/server/tests/lib/PreviewTest.php:408

30) Test\PreviewTest::testSecondPreviewsGetCachedMax with data set #29 (1, 729, 275, true, true)
Failed asserting that two strings are identical.
--- Expected
+++ Actual
@@ @@
-thumbnails/3457/756-537-max.png
+thumbnails/3457/756-538-max.png

/drone/src/github.com/nextcloud/server/tests/lib/PreviewTest.php:408

31) Test\PreviewTest::testSecondPreviewsGetCachedMax with data set #30 (1, -655, 461, true, true)
Failed asserting that two strings are identical.
--- Expected
+++ Actual
@@ @@
-thumbnails/3467/756-537-max.png
+thumbnails/3467/756-538-max.png

/drone/src/github.com/nextcloud/server/tests/lib/PreviewTest.php:408

32) Test\PreviewTest::testSecondPreviewsGetCachedMax with data set #31 (1, 590, -533, true, true)
Failed asserting that two strings are identical.
--- Expected
+++ Actual
@@ @@
-thumbnails/3477/756-537-max.png
+thumbnails/3477/756-538-max.png

/drone/src/github.com/nextcloud/server/tests/lib/PreviewTest.php:408

33) Test\PreviewTest::testDoNotCreatePreviewsLargerThanConfigMax with data set #6 (1, true, false)
Failed asserting that 538 matches expected 537.

/drone/src/github.com/nextcloud/server/tests/lib/PreviewTest.php:491

34) Test\PreviewTest::testDoNotCreatePreviewsLargerThanConfigMax with data set #7 (1, true, true)
Failed asserting that 538 matches expected 537.

/drone/src/github.com/nextcloud/server/tests/lib/PreviewTest.php:491

35) Test\PreviewTest::testIsBiggerWithAspectRatioCached with data set #4 (1, false, false)
Failed asserting that 356 matches expected 355.

/drone/src/github.com/nextcloud/server/tests/lib/PreviewTest.php:647
/drone/src/github.com/nextcloud/server/tests/lib/PreviewTest.php:543

36) Test\PreviewTest::testIsBiggerWithAspectRatioCached with data set #5 (1, false, true)
Failed asserting that 356 matches expected 355.

/drone/src/github.com/nextcloud/server/tests/lib/PreviewTest.php:647
/drone/src/github.com/nextcloud/server/tests/lib/PreviewTest.php:543

37) Test\PreviewTest::testIsBiggerWithAspectRatioCached with data set #6 (1, true, false)
Failed asserting that 285 matches expected 284.

/drone/src/github.com/nextcloud/server/tests/lib/PreviewTest.php:647
/drone/src/github.com/nextcloud/server/tests/lib/PreviewTest.php:540

38) Test\PreviewTest::testIsBiggerWithAspectRatioCached with data set #7 (1, true, true)
Failed asserting that 285 matches expected 284.

/drone/src/github.com/nextcloud/server/tests/lib/PreviewTest.php:647
/drone/src/github.com/nextcloud/server/tests/lib/PreviewTest.php:540
```

**Comment by rullzer** [comment]
Fix in https://github.com/nextcloud/server/pull/330

**Comment by MorrisJobke** [comment]
> Fix in #330

Merged
